photon-client/src/test/scala/com/linkedin/photon/ml/data/avro/NameAndTermFeatureMapUtilsTest.scala

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright 2017 LinkedIn Corp. All rights reserved.
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linkedin.photon.ml.data.avro
+
+import org.apache.hadoop.fs.Path
+import org.apache.spark.rdd.RDD
+import org.testng.Assert._
+import org.testng.annotations.{DataProvider, Test}
+
+import com.linkedin.photon.ml.test.{SparkTestUtils, TestTemplateWithTmpDir}
+
+/**
+ * Unit tests for the [[NameAndTermFeatureMapUtils]].
+ */
+class NameAndTermFeatureMapUtilsTest extends SparkTestUtils with TestTemplateWithTmpDir {
+
+  @DataProvider
+  def keyNameTermMap(): Array[Array[Any]] = {
+    val fixedEffect = Set(NameAndTerm("f1", "t1"), NameAndTerm("f1", "t2"), NameAndTerm("f2", "t3"))
+    val randomEffect = Set(NameAndTerm("f1", "t1"), NameAndTerm("f3", "t4"))
```

Comment:
A little confusing that 'f2' and 'f3' only have one instance in the data but their terms are 't3' and 't4'. I think it would be more consistent to make them both 't1'.